File: server/routers/board.py
"""板块行情路由"""
from fastapi import APIRouter, Query
from server.services import board_service

router = APIRouter(prefix="/board", tags=["板块行情"])


@router.get("/industry")
async def get_industry_boards(cache_minutes: int = Query(2, description="缓存分钟数")):
    """获取行业板块实时行情"""
    data = board_service.get_industry_boards(cache_minutes)
    return {"data": data, "total": len(data)}


@router.post("/industry/refresh")
async def refresh_industry_boards():
    """强制刷新行业板块行情"""
    return board_service.refresh_industry_boards()


@router.get("/concept")
async def get_concept_boards(cache_minutes: int = Query(2, description="缓存分钟数")):
    """获取概念板块实时行情"""
    data = board_service.get_concept_boards(cache_minutes)
    return {"data": data, "total": len(data)}


@router.post("/concept/refresh")
async def refresh_concept_boards():
    """强制刷新概念板块行情"""
    return board_service.refresh_concept_boards()


@router.get("/concept/{name}/kline")
async def get_concept_kline(
    name: str,
    period: str = Query('Y1', description="周期: Y1/Y3/Y5/ALL"),
    code: str = Query('', description="板块代码(platecode)，用于名称映射"),
):
    """获取概念板块指数 K 线"""
    return board_service.get_concept_kline(name, period, code)


@router.get("/watchlist")
async def get_board_watchlist():
    """获取关注的板块代码列表"""
    codes = board_service.get_watched_board_codes()
    return {"data": list(codes)}


@router.post("/watchlist/{code}")
async def add_board_watch(code: str, name: str = Query(..., description="板块名称")):
    """关注板块"""
    return board_service.add_board_watch(code, name)


@router.delete("/watchlist/{code}")
async def remove_board_watch(code: str):
    """取消关注板块"""
    return board_service.remove_board_watch(code)

File: web/src/api/board.ts
/** 板块行情 API */
import client from './client'

export interface IndustryBoard {
  code: string
  name: string
  stock_count: number
  avg_price: number
  change: number
  change_pct: number
  volume: number
  amount: number
  lead_stock_code: string
  lead_stock_price: number
  lead_stock_change: number
  lead_stock_change_pct: number
  lead_stock_name: string
}

/** 获取行业板块行情 */
export async function getIndustryBoards(cacheMinutes?: number): Promise<IndustryBoard[]> {
  const params: any = {}
  if (cacheMinutes != null) params.cache_minutes = cacheMinutes
  const { data } = await client.get('/board/industry', { params })
  return data.data
}

export interface ConceptBoard {
  code: string
  name: string
  change_pct: number
  net_inflow: number
  strength: number
  lead_stock_code: string
  lead_stock_name?: string
}

/** 强制刷新行业板块行情 */
export async function refreshIndustryBoards(): Promise<{ total: number; updated_at: string }> {
  const { data } = await client.post('/board/industry/refresh')
  return data
}

/** 获取概念板块行情 */
export async function getConceptBoards(cacheMinutes?: number): Promise<ConceptBoard[]> {
  const params: any = {}
  if (cacheMinutes != null) params.cache_minutes = cacheMinutes
  const { data } = await client.get('/board/concept', { params })
  return data.data
}

/** 强制刷新概念板块行情 */
export async function refreshConceptBoards(): Promise<{ total: number; updated_at: string }> {
  const { data } = await client.post('/board/concept/refresh')
  return data
}

/** 获取概念板块指数 K 线 */
export async function getConceptKline(
  name: string,
  period?: string,
  code?: string,
): Promise<{ name: string; data: any[] }> {
  const params: any = {}
  if (period) params.period = period
  if (code) params.code = code
  const { data } = await client.get(`/board/concept/${encodeURIComponent(name)}/kline`, { params })
  return data
}

/** 获取关注的板块代码列表 */
export async function getBoardWatchlist(): Promise<string[]> {
  const { data } = await client.get('/board/watchlist')
  return data.data
}

/** 关注板块 */
export async function addBoardWatch(code: string, name: string): Promise<void> {
  await client.post(`/board/watchlist/${encodeURIComponent(code)}`, null, { params: { name } })
}

/** 取消关注板块 */
export async function removeBoardWatch(code: string): Promise<void> {
  await client.delete(`/board/watchlist/${encodeURIComponent(code)}`)
}
